src/redux/books/books.js

```diff
@@ -1,27 +1,47 @@
+import { getAllBooks, postBook, deleteBook } from '../../api/api';
+
 const ADD_BOOK = 'bookStore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+const GET_BOOKS = 'bookStore/books/GET_BOOKS';
 
 const initialState = [];
 
-export const addBook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
-});
+const convertDataObjectToArray = (data) => {
+  const dataArray = [];
+
+  Object.keys(data).map((key) => {
+    const book = data.key[0];
+    book.item_id = key;
+    return dataArray.push(book);
+  });
 
-export const removeBook = (payload) => ({
-  type: REMOVE_BOOK,
-  payload,
-});
+  return dataArray;
+};
+
+export const getBooks = () => async (dispatch) => {
+  const data = await getAllBooks();
+  const convertedData = convertDataObjectToArray(data);
+  dispatch({ type: GET_BOOKS, convertedData });
+};
+
+export const addBook = (payload) => async (dispatch) => {
+  await postBook(payload);
+  dispatch({ type: ADD_BOOK, payload });
+};
+
+export const removeBook = (payload) => async (dispatch) => {
+  await deleteBook(payload);
+  dispatch({ type: REMOVE_BOOK, payload });
```

Comment:
Please make `postBook` and `removeBook` server requests with react hooks. All the server requests should be made in the componentDidMount lifecycle method or with react hooks.